0104_MaximumDepthOfBinaryTree.py

```diff
@@ -0,0 +1,47 @@
+class TreeNode(object):
+    def __init__(self, x):
+        self.val = x
+        self.left = None
+        self.right = None
+
+class Solution(object):
+    def maxDepth(self, root):
+        currMaxDepth = 0
+        currRoot = root
+        if currRoot != None:
+            if (self.maxDepth(root.left)>self.maxDepth(root.right)):
+                currMaxDepth = self.maxDepth(root.left)+1
```

Comment:
This is not good, as in line 12 you are calling recursive call for both left and right side, but in 13 or 15 you again call for left or right depending on which one is bigger. That said, recursive calls (which are expensive) can be avoided if you store these values in temp variables.